include/hlslib/xilinx/OpenCL.h

```diff
@@ -28,10 +28,59 @@ namespace ocl {
 
 #if HLSLIB_LEGACY_SDX == 0
 constexpr auto kXilinxMemPointer = CL_MEM_EXT_PTR_XILINX;
-constexpr auto kMemoryBank0 = XCL_MEM_DDR_BANK0;
-constexpr auto kMemoryBank1 = XCL_MEM_DDR_BANK1;
-constexpr auto kMemoryBank2 = XCL_MEM_DDR_BANK2;
-constexpr auto kMemoryBank3 = XCL_MEM_DDR_BANK3;
+
+//constexpr auto kMemoryBank0 = 32 | XCL_MEM_TOPOLOGY;
+//constexpr auto kMemoryBank1 = 33 | XCL_MEM_TOPOLOGY;
+
+class XilinxDDRBankFlags {
+public:
+  XilinxDDRBankFlags(const std::string &device_name) {
+    if(device_name == "xilinx_u280_xdma_201920_3") {
```

Comment:
I think we should match _any_ U280 DSA, not just `xdma_201920_3` (use `std::string::find`).